src/System.Reflection.Emit/tests/GenericTypeParameterBuilder/GenericTypeParameterBuilderGUID.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Reflection;
+using System.Reflection.Emit;
+using Xunit;
+
+namespace System.Reflection.Emit.Tests
+{
+    public class GenericTypeParameterBuilderGUID
+    {
+        [Fact]
+        public void NegTest1()
+        {
+            AssemblyName myAsmName = new AssemblyName("GenericEmitExample1");
+            AssemblyBuilder myAssembly = AssemblyBuilder.DefineDynamicAssembly(myAsmName, AssemblyBuilderAccess.Run);
+            ModuleBuilder myModule = TestLibrary.Utilities.GetModuleBuilder(myAssembly, myAsmName.Name);
+
+            Type baseType = typeof(ExampleBase);
+
+            TypeBuilder myType = myModule.DefineType("Sample", TypeAttributes.Public);
+
+            string[] typeParamNames = { "TFirst" };
+            GenericTypeParameterBuilder[] typeParams = myType.DefineGenericParameters(typeParamNames);
+
+            GenericTypeParameterBuilder TFirst = typeParams[0];
+            Assert.Throws<NotSupportedException>(() => { Guid guid = TFirst.GUID; });
```

Comment:
Yes, i agree, but this is just a different way of coding, and considering these are legacy tests, and there are going to be lots of instances like this, i dont think it's worth the effort to write it in a particular way, because the test is not for lambda expressions.